dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/programrule/DefaultProgramRuleActionService.java

```diff
@@ -96,4 +96,25 @@ public List<ProgramRuleAction> getProgramRuleActionsWithNoStageId() {
     return programRuleActionStore.getMalFormedRuleActionsByType(
         ProgramRuleActionType.HIDEPROGRAMSTAGE);
   }
+
+  @Override
+  @Transactional(readOnly = true)
+  public List<String> getProgramStagesPresentInProgramRuleActions() {
```

Comment:
This method is specific for `SCHEDULEDEVENT` action type, could we put it in the name so we also remove the need for a comment in the dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/TrackerIdentifierCollector.java?